ext/pdo/pdo_dbh.c

```diff
@@ -680,9 +685,8 @@ static int pdo_dbh_attribute_set(pdo_dbh_t *dbh, zend_long attr, zval *value) /*
 	zend_long lval;
 
 #define PDO_LONG_PARAM_CHECK \
-	if (Z_TYPE_P(value) != IS_LONG && Z_TYPE_P(value) != IS_STRING && Z_TYPE_P(value) != IS_FALSE && Z_TYPE_P(value) != IS_TRUE) { \
-		pdo_raise_impl_error(dbh, NULL, "HY000", "attribute value must be an integer"); \
-		PDO_HANDLE_DBH_ERR(); \
+	if (Z_TYPE_P(value) != IS_LONG && Z_TYPE_P(value) != IS_FALSE && Z_TYPE_P(value) != IS_TRUE) { \
```

Comment:
Looks like this dropped IS_STRING?

Reply:
I did because it accepted values such as `'nonsense'` for integer values (as can be seen by some of the test changes which silently accepted this before). I can revert this specific part again.

Reply:
I would probably revert it. The problem is that we don't have the distinction between `"42"` and `"nonesense"` here, and we don't want to break the former case without warning, I think?